### ISSUE TITLE
Arby improving suggestions

### ISSUE THREAD

**Issue by raladev** [ExchangeUnion/market-maker-tools#16]
Arby is in alpha state now and looks unfinished. So, here you can offer any improvement for discussion.

- [ ] if invalid creds are passed to arby, status command should contain message about that, also logs of arby should contain error about bad creds and maybe container should finish work with another exit code (not 0).

- [ ] Add info rder that was sent to binance + also i want to see info about binance fill.

- [ ] Maybe it would be better to do not print same price twice or move printing all prices to lower debug level.

- [ ] bestBid and bestAsk monitoring looks like more correct solution then last trade monitoring.

- [ ] Current behavior can be destructive if price will move significantly after binance order placement, if I understood correctly. OpenDex update was stopped by placed but not filled order, so xud node will contain orders with tasty prices.
```
- is it true that each Opendex trade leads to order canceling untill Binance fill?
- - I haven't implemented this logic yet. Right now it just stops updating orders until Binance order is in progress. It should probably keep updating them or remove all orders until centralized exchange part is done.
```

- [ ] Orders should be canceled if Arby completes its work (for example, because of empty channel of shutting down by user)

- [ ] Let's fix the order update interval to 5s, at the moment orders are getting replaced that fast that our matching routine can't keep up:

```
simnet > buy 1 eth/btc mkt
matched 1 ETH @ 0.024507483 with peer ConsiderMoral order 039a7cc0-aef8-11ea-9144-37681cbd4116, attempting swap...
failed to swap 1 ETH due to OrderNotFound with peer order 039a7cc0-aef8-11ea-9144-37681cbd4116, continuing with matching routine...
matched 1 ETH @ 0.024505481 with peer ConsiderMoral order 03c3fdc0-aef8-11ea-9144-37681cbd4116, attempting swap...
failed to swap 1 ETH due to UnknownError with peer order 03c3fdc0-aef8-11ea-9144-37681cbd4116, continuing with matching routine...
matched 1 ETH @ 0.024503479 with peer ConsiderMoral order 087ec6b0-aef8-11ea-9144-37681cbd4116, attempting swap...
successfully swapped 1 ETH with peer order 087ec6b0-aef8-11ea-9144-37681cbd4116
```

- [ ] Another thing: you are always using `replace_order_id` when issuing new orders, right? I am confused by local xud/arby logs saying `removed order` and creating new orders, but but there were no remova p2p packets. In that case, I'd remove this arby log statement since it's confusing: `[OpenDEX] trace: Removed all open orders for ETH/BTC`

**Comment by raladev** [comment]
> * Another thing: you are always using `replace_order_id` when issuing new orders, right? I am confused by local xud/arby logs saying `removed order` and creating new orders, but but there were no remova p2p packets. In that case, I'd remove this arby log statement since it's confusing: `[OpenDEX] trace: Removed all open orders for ETH/BTC`

if there is no real removal/creation then it should be:

1. 'Create order' msg when repalce_id is not used (first time place)
2 'Order replace' msg when raplce_id is used

**Comment by raladev** [comment]
- [ ] it would be good to have cli tool for arby to control it.

For now it is uncontrollable container that starts order placing when see balance in the channel.

But ut would be good to have at least start/stop commands in docker setup.

**Comment by kilrau** [comment]
>     * [ ]  it would be good to have cli tool for arby to control it.
> 
> 
> For now it is uncontrollable container that starts order placing when see balance in the channel.
> But it would be good to have at least start/stop commands in docker setup.

Good idea, but I would rather let xud ctl take care of that: warn & educate user before starting. Offer a command to stop it etc.

**Comment by raladev** [comment]
- [ ] qty of orders should be based on channel balance of both currencies
Arby places sell 130 ETH/BTC 0.024499475 order, but I have only 7 ETH in the channel, so it can lead to significant amount of errors during swap
[Screenshot from 2020-06-15 19-11-07](https://user-images.githubusercontent.com/29906866/84680843-fe252580-af3b-11ea-9dd3-1f8a9065a2c8.png)

**Comment by kilrau** [comment]
Marked above as done, it has its own P1 issue and will be done asap: https://github.com/ExchangeUnion/market-maker-tools/issues/17

**Comment by kilrau** [comment]
I think we should tackle this one in course of this issue next:
>  - bestBid and bestAsk monitoring looks like more correct solution then last trade monitoring.
